ipam/actor.go

```diff
@@ -0,0 +1,165 @@
+package ipam
+
+import (
+	"fmt"
+	"github.com/zettio/weave/router"
+	"net"
+	"time"
+)
+
+// Start runs the allocator goroutine
+func (alloc *Allocator) Start() {
+	actionChan := make(chan func(), router.ChannelSize)
+	alloc.actionChan = actionChan
+	go alloc.actorLoop(actionChan)
+}
+
+// Actor client API
+
+// GetFor (Sync) - get IP address for container with given name
+func (alloc *Allocator) GetFor(ident string, cancelChan <-chan bool) net.IP {
+	resultChan := make(chan net.IP, 1) // len 1 so actor can send while cancel is in progress
```

Comment:
you could keep this at 0 by making cancellation read from resultChan, and the cancellation handler send nil to the resultChan if it finds the id.